model/assigned_vm_address.rb

```diff
@@ -5,6 +5,11 @@
 class AssignedVmAddress < Sequel::Model
   one_to_one :vm, key: :dst_vm_id
   many_to_one :address, key: :address_id
+  one_to_one :active_billing_record, class: :BillingRecord, key: :resource_id, conditions: {Sequel.function(:upper, :span) => nil}
 
   include ResourceMethods
+
+  def before_destroy
```

Comment:
This needs to call super, so that it doesn't overwrite before_destroy in model.rb